File: ios/Poke/Services/Sources/Services/Protocols/FavoriteService.swift
//
//  File.swift
//  
//
//  Created by Dylan Velez on 10/08/21.
//

import Foundation
import Combine
import Models

public protocol FavoriteService {
    func fetchFavorites() -> Set<Int>
    func fetchList() -> AnyPublisher<Pets, Error>
    func addToFavorites(id: Int)
    func removeFromFavorites(id: Int)
}

File: android/Pets-Starter/app/src/main/java/com/pets/services/Interfaces/FavoriteService.kt
package com.pets.services.Interfaces

import com.pets.models.Pet
import retrofit2.http.GET

interface FavoriteService {
    fun fetchFavorites(): Set<Int>
    @GET(".")
    suspend fun fetchList(): List<Pet>
    fun addToFavorites(id: Int)
    fun removeFromFavorites(id: Int)
}